app/src/main/java/com/example/tasky/common/model/ResolvableResource.kt

```diff
@@ -0,0 +1,22 @@
+package com.example.tasky.common.model
+
+import android.content.Context
+import androidx.annotation.StringRes
+
+interface ResolvableResource {
+    fun resolve(context: Context): String
+}
+
+data class ResolvableString(val string: String) : ResolvableResource {
+    override fun resolve(context: Context): String {
+        return string
+    }
+}
+
+data class ResolvableStringRes(
+    @StringRes val stringRes: Int
+) : ResolvableResource {
+    override fun resolve(context: Context): String {
+        return context.getString(stringRes)
+    }
+}
```

Comment:
Could it make sense to make `ResolvableResource` a `sealed interface` to make it exhaustive when checking for its type?